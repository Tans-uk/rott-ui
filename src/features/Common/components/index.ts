export * from './CommonItem'
export * from './CommonItemContainer'
