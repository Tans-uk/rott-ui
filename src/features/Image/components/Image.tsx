import {type FC} from 'react'

import {
  Image as RNImage,
  StyleSheet,
  type ImageSourcePropType,
  type ImageURISource,
} from 'react-native'

import type {ImageProps} from '../models'
import {ImageStyles} from '../styles'

import {Images} from '@constants'

export const Image: FC<ImageProps> = ({variant, size, style, source, name, ...props}) => {
  const imageSource = Object.entries(Images).find(
    (image) => image[0]?.toLowerCase() === name?.toLowerCase()
  )

  const uriImageSource = {...(source as ImageURISource), cache: 'reload'} as ImageSourcePropType

  return (
    <RNImage
      style={StyleSheet.flatten([ImageStyles({variant, size, ...props}).defaultImageStyle, style])}
      source={source ? uriImageSource : Images[imageSource?.[0] as never]}
      {...props}
    />
  )
}
