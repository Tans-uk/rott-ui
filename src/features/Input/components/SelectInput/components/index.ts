export * from './SelectInput'
export * from './SelectInputModalComponent'
