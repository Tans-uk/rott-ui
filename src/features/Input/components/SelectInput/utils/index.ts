export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
export * from './sortListBySearchPriority'
