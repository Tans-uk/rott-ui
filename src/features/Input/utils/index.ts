export * from './formatByDateMode'
export * from './useInputStyleNormalizer'
