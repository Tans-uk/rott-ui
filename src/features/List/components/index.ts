export * from './List'
export * from './FavoriteSwipeComponent'
export * from './ListSkeletonItem'
