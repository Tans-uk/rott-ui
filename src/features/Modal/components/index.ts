export * from './ModalComponent'
export * from './ModalContainer'
