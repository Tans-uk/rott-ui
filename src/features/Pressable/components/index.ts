export * from './Pressable'
export * from './PressableContainer'
