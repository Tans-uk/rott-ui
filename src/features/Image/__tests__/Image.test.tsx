import {render} from '../../../__tests__/utils/testUtils'
import {Image} from '../components'
import type {ImageTypes} from '../models'

describe('Image -> Custom Component', () => {
  const testId = {
    imageTestId: 'image-test-id',
  }
  const testIconName: ImageTypes = 'PTTBANK_WHITE'

  it('image elementi snapshot ile eşleşmeli', () => {
    const {imageTestId} = testId
    const renderedIcon = render(
      <Image testID={imageTestId} name={testIconName} width={50} height={50} />
    )

    expect(renderedIcon).toMatchSnapshot()
  })

  it('image elementi renderlandığında gerçekten ekranda olmalı', () => {
    const {imageTestId} = testId
    const {getByTestId} = render(<Image testID={imageTestId} name={testIconName} width={50} />)

    const imageElement = getByTestId(imageTestId)

    expect(imageElement).toBeOnTheScreen()
  })

  it('image elementi verilen width ile renderlanmalı', () => {
    const {imageTestId} = testId
    const {getByTestId} = render(<Image testID={imageTestId} name={testIconName} width={50} />)

    const imageElement = getByTestId(imageTestId)

    expect(imageElement).toHaveProp('width', 50)
  })

  it('image elementi verilen height ile renderlanmalı', () => {
    const {imageTestId} = testId
    const {getByTestId} = render(<Image testID={imageTestId} name={testIconName} height={50} />)

    const imageElement = getByTestId(imageTestId)

    expect(imageElement).toHaveProp('height', 50)
  })
})
