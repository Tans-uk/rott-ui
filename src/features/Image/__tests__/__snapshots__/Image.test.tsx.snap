// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image -> Custom Component image elementi snapshot ile eşleşmeli 1`] = `
<KeyboardProvider
  enabled={true}
>
  <View
    flex={1}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Image
      height={50}
      source={
        {
          "testUri": "../../../src/assets/images/logo/pttbank-White.png",
        }
      }
      style={
        {
          "height": 96,
          "maxHeight": 96,
          "overflow": "hidden",
          "resizeMode": "contain",
          "width": 96,
        }
      }
      testID="image-test-id"
      width={50}
    />
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        [
          {
            "alignSelf": "center",
            "elevation": 999999,
            "flex": 0,
            "maxWidth": "100%",
            "position": "absolute",
            "zIndex": 999999,
          },
          {
            "flexDirection": "column-reverse",
            "justifyContent": "flex-start",
            "top": 30,
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          {
            "bottom": 0,
          }
        }
      >
        <RCTSafeAreaView />
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        [
          {
            "alignSelf": "center",
            "elevation": 999999,
            "flex": 0,
            "maxWidth": "100%",
            "position": "absolute",
            "zIndex": 999999,
          },
          {
            "bottom": 40,
            "flexDirection": "column",
            "justifyContent": "flex-end",
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          {
            "bottom": 0,
          }
        }
      >
        <RCTSafeAreaView />
      </View>
    </View>
  </View>
</KeyboardProvider>
`;
